src/orquestra/sdk/_base/_logs/_markers.py

```diff
@@ -0,0 +1,136 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+import json
+import re
+import sys
+import typing as t
+from contextlib import contextmanager
+from dataclasses import dataclass
+
+from orquestra.sdk.schema.ir import TaskInvocationId
+from orquestra.sdk.schema.workflow_run import WorkflowRunId
+
+from .. import _dates
+
+ORQ_MARKER_PREFIX = "ORQ-MARKER:"
+ORQ_MARKER_PATTERN = re.compile(re.escape(ORQ_MARKER_PREFIX) + r"(.+)")
+
+
+# `TaskStartMarker` and `TaskEndMarker` are record types that hold the information we
+# emit in the log markers. They're dataclasses and not pydantic models because we're
+# not just using plain JSON: we're using prefix markers.
+
+
+@dataclass(frozen=True)
+class TaskStartMarker:
```

Comment:
What would you suggest to put in the docstring for this dataclass? The only thing I had in mind was along the lines of "Represents a marker we put at the end of a task" but this would convey no additional information, and _I'm not a fan_ of noise.

Reply:
Fair point, it's a tension between disliking noise and disliking undocumented classes. Let's leave it as is, and revisit should should the class ever grow more complicated!